metrics/web/docs/03_widget_structure_organization.md

```diff
@@ -1,54 +1,350 @@
+
+5. If widget contains any constant strings like titles, descriptions, error messages, and so on, consider extracting them to a separate class in the `strings` folder.
+
+The following diagram describes the process of creation of the high-level widget:
+
+![Create High-Level Widget Diagram](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/software-platform/monorepo/widget_stucture_organization_document/metrics/web/docs/diagrams/create_high_level_widget_activity_diagram.puml)
+
+The next question we should answer is: "Should we create a separate widget for each UI component?".
+For example, we have one low-level widget that displays the circular percentage chart, and we have two metrics that should be displayed with this chart. The question is - should we create a separate widget for each of these metrics.
+
+Let's consider the pros & cons of creating a separate widget for each metric: 
+
+Pros: 
+- SRP: one widget for one view.
+- It makes the namings a bit more intuitive.
+- Reduces the number of changes to make one widget looks different.
+
+Cons: 
+- Increases code duplication in implementation and tests.
+- Increases the amount of code.
+
+Another way is to create a single configurable high-level widget for displaying these metrics. Let's consider the pros & cons of this approach:
+
+Pros: 
+- Keeps your code DRY.
+- It does not violate the single responsibility principle.
+- Reduces the amount of code and thus reduces the number of possible bugs, errors, etc.
+
+Cons: 
+- Makes your code less maintainable.
+- Increases amount of changes to make one widget looks different.
+
+So, it seems to be better to create a separate widget for each view even if these widgets look identical currently. It will allow us to simply change one of them later and increase maintainability.
 
 ## Metrics Theme guidelines
+
+### Metrics Theme structure [Pending Approval]
 > Explain and diagram the metrics theme structure.
 
+In order to support the application's appearance in a clear and maintainable way, the Metrics Web Application uses themes. The theme is a configuration for the appearance (colors, text style, etc.) of a part of UI. Themes allow configuring how the UI or its part looks like and introduces the great support to change this appearance reactively (e.g. dark and light themes).
+
+Our main concept of the theme is that we have a `MetricsThemeData` class that contains all the theme data for widgets. We can obtain the `MetricsThemeData` using the `MetricsTheme.of(context)` method. The `MetricsThemeData` consists of classes that implement the  `AttentionLevelThemeData` interface. This interface provides an `attentionLevel` field that provides the different variations of styles, applicable for separate group of widgets. Let's consider the class diagram that represents structure of `MetricsThemeData` and the relationships between classes in the theme data and widgets: 
```

Comment:
Should we move this below (after structure) etc... and place it into the section "How to use (get) theme"?